great_expectations/datasource/data_connector/configured_asset_sql_data_connector.py

```diff
@@ -245,13 +271,36 @@ def _generate_batch_spec_parameters_from_batch_definition(
             dict built from batch_definition
         """
         data_asset_name: str = batch_definition.data_asset_name
+        table_name: str = self._get_table_name_from_batch_definition(batch_definition)
         return {
             "data_asset_name": data_asset_name,
-            "table_name": data_asset_name,
+            "table_name": table_name,
             "batch_identifiers": batch_definition.batch_identifiers,
             **self.assets[data_asset_name],
         }
 
+    def _get_table_name_from_batch_definition(
+        self, batch_definition: BatchDefinition
+    ) -> str:
+        """
+            Helper method called by _get_batch_identifiers_list_from_data_asset_config() to parse table_name from data_asset_name in cases
+            where schema is included.
+
+            data_asset_name in those cases are [schema].[table_name].
+
+        function will split data_asset_name on [schema]. and return the resulting table_name.
+        """
+        table_name: str = batch_definition.data_asset_name
+        if "schema_name" in self.assets[batch_definition.data_asset_name]:
+            schema_name_str: str = self.assets[batch_definition.data_asset_name][
+                "schema_name"
+            ]
```

Comment:
can we assign `self.assets[batch_definition.data_asset_name]` to a var so we can reuse?

Reply:
done 🙇🏼  Thank you... it's *much* better.